components/sync15/src/client.rs

```diff
@@ -43,40 +103,56 @@ pub struct Sync15StorageClient {
 }
 
 impl SetupStorageClient for Sync15StorageClient {
-    fn fetch_info_configuration(&self) -> error::Result<InfoConfiguration> {
-        let server_config = self.fetch_info::<InfoConfiguration>("info/configuration")?;
-        Ok(server_config)
+    fn fetch_info_configuration(&self) -> error::Result<Sync15ClientResponse<InfoConfiguration>> {
+        self.relative_storage_request(Method::Get, "info/configuration")
     }
 
-    fn fetch_info_collections(&self) -> error::Result<InfoCollections> {
-        let collections = self.fetch_info::<InfoCollections>("info/collections")?;
-        Ok(collections)
+    fn fetch_info_collections(&self) -> error::Result<Sync15ClientResponse<InfoCollections>> {
+        self.relative_storage_request(Method::Get, "info/collections")
     }
 
-    fn fetch_meta_global(&self) -> error::Result<BsoRecord<MetaGlobalRecord>> {
-        let resp = match self.relative_storage_request(Method::Get, "storage/meta/global") {
-            Ok(r) => Ok(r),
-            Err(ref e) if e.is_not_found() => Err(ErrorKind::NoMetaGlobal.into()),
-            Err(e) => Err(e),
-        }?;
-        // Note: meta/global is not encrypted!
-        let meta_global: BsoRecord<MetaGlobalRecord> = resp.json()?;
-        log::trace!("Meta global: {:?}", meta_global.payload);
-        Ok(meta_global)
+    fn fetch_meta_global(&self) -> error::Result<Sync15ClientResponse<MetaGlobalRecord>> {
+        // meta/global is a Bso, so there's an extra dance to do.
+        let got: Sync15ClientResponse<BsoRecord<MetaGlobalRecord>> =
+            self.relative_storage_request(Method::Get, "storage/meta/global")?;
+        Ok(match got {
+            Sync15ClientResponse::Success {
+                record,
+                last_modified,
+                route,
+            } => Sync15ClientResponse::Success {
+                record: record.payload,
+                last_modified,
+                route,
+            },
+            Sync15ClientResponse::NotFound { route } => Sync15ClientResponse::NotFound { route },
```

Comment:
Can you have a catch-all `response => response` here, that passes error responses through as-is?

Reply:
I actually tried that, and was surprised-not-surprised to find it doesn't work - the `<T>` in `Sync15ClientResponse<T>`  is different, so the rust compiler treats them as completely different types.